src/Corax/IndexEntryBuilder.cs

```diff
@@ -124,6 +127,33 @@ public void Write(int field, ReadOnlySpan<byte> value)
 
             _dataIndex += length + value.Length;
         }
+        
+        public void Write(int field, Span<byte> blittableBinary, Span<byte> blittableAsText)
+        {
+            //STRUCT
+            //<type><size_of_binary><size_of_text><binary><text>
+            Debug.Assert(field < _knownFields.Count);
+            Debug.Assert(_knownFieldsLocations[field] == Invalid);
+            
+            if (blittableBinary.Length == 0)
+                return;
+
+            int dataLocation = _dataIndex;
+            // Write known field.
+            _knownFieldsLocations[field] = dataLocation | unchecked((int)0x80000000);
+            dataLocation += VariableSizeEncoding.Write(_buffer, (byte)IndexEntryFieldType.Blittable, dataLocation);
+            dataLocation += VariableSizeEncoding.Write(_buffer, blittableBinary.Length, dataLocation);
+            dataLocation += VariableSizeEncoding.Write(_buffer, blittableAsText.Length, dataLocation);
```

Comment:
Why do we have that as text? Why store that twice?

Reply:
Because some values are encoded in raw memory of blittable and applying analyzers on it will cause losing them. We can pass BlittableReader and JsonContext into Indexing inside Corax (to convert Blittable into JSON) but it's will cause a mix of abstraction levels.